### ISSUE TITLE
Ember Data v1.0.0-beta.3

### ISSUE THREAD

**Issue by seanwilson** [eltiare/ember-websql-storage#1]
This is the only Websql adapter I've found for Ember - great work!

I'm wondering if you're going to update it to work with the latest versions of Ember data and Ember as I'm having a couple of issues..

Let me know if you need more info.  Really hope you're still working on it!

**Comment by eltiare** [comment]
Yup, I plan on updating it to the most recent version of Ember by next week. Bug me if you don't see anything by next Friday.

**Comment by seanwilson** [comment]
Wow, that is truly awesome news!!! :)))

I've been struggling for a few days, but it's got me beat.

I'd offer to help but I'm new to Ember - I've been looking for a more
structured way to build Phonegap apps, and I'm hoping that once this piece
of the jigsaw is in place, Ember and Web Sql will be the way forward.

I'll keep a lookout, and yes of course - will give you a nudge if I don't
see anything within a week or so!

Thanks again, and thanks for the swift reply! :)

Sean

**Comment by eltiare** [comment]
Just pushed some code - the tests are returning weird errors but that may just have something to do with the funky way Ember+Data acts in the testing environment. Seems like everything has to be loaded manually. Give it a go and let me know if you run into any errors.

**Comment by seanwilson** [comment]
Hi Jeremy, sorry for the delay in getting back to you..

I'm getting a couple of errors too. I've attached a screenshot of the
output in Chrome Version 30.0.1599.101 on OSX.

I've also attached the output from the debug console.

As a side note, I have the Ember plugin for Chrome and that was saying "Ember
application not detected", although I did note that you've changed
the window.App declaration from Ember.Application.create() - presumably as
a temporary measure to get started.

Let me know if I can help any more :)

Thanks,

Sean

**Comment by eltiare** [comment]
Are you having the errors in the tests or in your app? I'm having the hardest time trying to set up Ember for QUnit testing - there seems to be all sorts of Voodoo in order to test components like the adapters. I'm not seeing the attachment here on Github - please send it via a gist (gist.github.com) or try it again.

**Comment by seanwilson** [comment]
I'm getting the errors in the test file - I haven't tried it with my app yet. I've attached the screenshots to this thread, they were sent by email before so that's probably why they've not posted.

![screen shot 2013-10-21 at 11 56 27](https://f.cloud.github.com/assets/416598/1376415/168670ee-3a94-11e3-99ea-9b4b1dee9d40.png)
![screen shot 2013-10-21 at 11 55 16](https://f.cloud.github.com/assets/416598/1376416/1689b326-3a94-11e3-9694-5d535ddac5de.png)

**Comment by eltiare** [comment]
Yup. Same errors I'm getting in testing. The error is buried deep in Ember code so I'm thinking it's a problem with my test setup. It may work in a real Ember app.

**Comment by eltiare** [comment]
The reason (I'm guessing) that I'm having a hard time with the tests is that the new Ember Data is coupled with the controllers - which really sucks if I just want to test the adapter like I do now. Hopefully I can figure out how to get the tests working here soon.

**Comment by eltiare** [comment]
Good news: I've installed enough hacks to make this sort of work. Still fighting though.

**Comment by seanwilson** [comment]
Nice work, thanks for the update. I'll give it a try.

**Comment by seanwilson** [comment]
I haven't had chance to implement it into my project, but I ran your test and got a couple of errors. Thought I'd attach screenshots if that helps.

Hoping to try it out with my code later :)

Nice work so far!

![screen shot 2013-10-25 at 14 00 07](https://f.cloud.github.com/assets/416598/1407848/da177abe-3d76-11e3-8da1-32621f165e46.png)
![screen shot 2013-10-25 at 14 01 03](https://f.cloud.github.com/assets/416598/1407849/de2a1d78-3d76-11e3-9c6f-777ad5d15f12.png)

**Comment by eltiare** [comment]
I'm thinking that ember-data is just too buggy to use right now. The biggest blocker right now is that ember-data likes to assign IDs before a record is even created - thus the ID for the model and the ID in the database will clash. I've spent way too much time trying to make this work for the scant documentation that's available. I'll give it another go once ember-data reaches official 1.0 status. Hopefully by then it will actually work for this type of data store and they have docs that match the reality of the code.

**Comment by seanwilson** [comment]
No worries, I totally get what you mean. I've been struggling with the documentation myself as some places say one thing and others say something totally different. Hopefully soon they'll release a stable version with consistent documentation. 

Really appreciate your efforts though!

Back to the drawing board for now...

**Comment by simontseng** [comment]
great work!

just tried your adapter and having the same error

however, i tried to comment out the serializer and default it to null like in the FixtureAdapter then the "type error" is gone.  It let me get further along, before i hit other errors.  so do you think it is related to the JSONSerializer

**Comment by eltiare** [comment]
Sadly I can't sink any more time into this until the code reaches 1.0 status. There are too many things that can change.
